### DISCUSSION TITLE
Mobile Admin Accounts

### DISCUSSION THREAD

**Discussion by kris-fh** [Homebrew/discussions#1005] [Everyday usage]
Has anyone else noticed that Homebrew doesn't seem to play nice with Mobile Admin Accounts, that is user accounts that are created from AD network accounts as mobile accounts and given Admin rights. And you get a mix of inconsistent error messages, from "user not in Sudos file" to error "initializing audit plugin sudoers_audit".  Explicitly adding the user to the sudo file doesn't seem to fix and even if it did the next upgrade of Big Sur would move this altered file out the way.  Any ideas or help would be great...

**Comment by SMillerDev** [comment]
I think the solution here is making sure they are full administrator accounts. But this is more an active directory than a brew question.

**Reply by kris-fh** [reply]
I didn't realise macOS has levels of administrator?  I thought you were either an administrator or not regardless of if it was the local directory or a network directory that authenticated you as how every other macOS app works and Swift makes a provision to check. In fact, Mobile Users are a special kind of user who can be authenticated by both directories. I would have thought if Brew isn't able to recognise the admin rights of anything other than users who can be authenticated only by a local directory exclusively without errors then it would be a brew question?  It's my speculation that brew is making linux/unix like checks and not really checking anything with the macOS Open Directory implementation (the local directory) to correctly look at the user's privileges.

**Reply by SMillerDev** [reply]
Yeah, macOS uses Unix system standards to check for permissions because it needs them to run Unix commands. If macOS doesn't manage those correctly there is nothing brew can do.

**Reply by kris-fh** [reply]
It doesn't that was my point was more that macOS has progressed well beyond the old unix/linux methods and uses the aforementioned local Open Directory variant. Brew is using (I speculate) these old methods and as a result, Brew doesn't know how to work correctly with this type of user.  That is to say that OS has evolved in a more advanced manner and it's time to play catch up on Brew's part should I be correct.  How much directory experience do you have? I'm finding it hard to make sense of what you're saying... and you seem not keen to help on this issue having shut it down twice...

**Reply by SMillerDev** [reply]
I have 0 active directory experience and IMHO using AD managed users isn't within the scope of the brew project to support if Apple doesn't want to support it in the Unix standards brew uses.

**Reply by kris-fh** [reply]
You know that an AD user is no different to a local user in this example, in fact, that's close to how it's designed?  And you're just talking about where the directory is hosted and the user type incorrectly?  Anyways best of luck managing the scope of your project as Apple moves more and more into the local OD variant and then starts to protect the OS more and of these old tools from being touched by the users and developers alike and moving the files needed to manage this a read-only system volume.

Great experience speaking with the brew team.
